src/connectors/index.ts

```diff
@@ -1,49 +1,157 @@
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-})
-
-export const gnosisSafe = new SafeAppConnector()
-
-export const walletconnect = new WalletConnectConnector({
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-  rpc: INFURA_NETWORK_URLS,
-  qrcode: true,
-})
-
-// mainnet only
-export const fortmatic = new FortmaticConnector({
-  apiKey: FORMATIC_KEY ?? '',
-  chainId: 1,
-})
-
-export const walletlink = new WalletLinkConnector({
-  url: INFURA_NETWORK_URLS[SupportedChainId.MAINNET],
-  appName: 'Uniswap',
-  appLogoUrl: UNISWAP_LOGO_URL,
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-})
+
+export enum Wallet {
+  INJECTED = 'INJECTED',
+  COINBASE_WALLET = 'COINBASE_WALLET',
+  WALLET_CONNECT = 'WALLET_CONNECT',
+  FORTMATIC = 'FORTMATIC',
+  NETWORK = 'NETWORK',
+  GNOSIS_SAFE = 'GNOSIS_SAFE',
+}
+
+export type ModalWallet = Exclude<Wallet, Wallet.NETWORK | Wallet.GNOSIS_SAFE>
+
+export const MODAL_WALLETS = [Wallet.COINBASE_WALLET, Wallet.WALLET_CONNECT, Wallet.INJECTED, Wallet.FORTMATIC]
+
+export const isChainAllowed = (connector: Connector, chainId: number) => {
+  switch (connector) {
+    case fortmatic:
+      return chainId === SupportedChainId.MAINNET
+    case injected:
+    case coinbaseWallet:
+    case walletConnect:
+    case network:
+    case gnosisSafe:
+      return ALL_SUPPORTED_CHAIN_IDS.includes(chainId)
+    default:
+      return false
+  }
+}
+
+export const getWalletForConnector = (connector: Connector) => {
+  switch (connector) {
+    case injected:
+      return Wallet.INJECTED
+    case coinbaseWallet:
+      return Wallet.COINBASE_WALLET
+    case walletConnect:
+      return Wallet.WALLET_CONNECT
+    case fortmatic:
+      return Wallet.FORTMATIC
+    case network:
+      return Wallet.NETWORK
+    case gnosisSafe:
+      return Wallet.GNOSIS_SAFE
+    default:
+      throw Error('unsupported connector')
+  }
+}
+
+export const getConnectorForWallet = (wallet: Wallet) => {
+  switch (wallet) {
+    case Wallet.INJECTED:
+      return injected
+    case Wallet.COINBASE_WALLET:
+      return coinbaseWallet
+    case Wallet.WALLET_CONNECT:
+      return walletConnect
+    case Wallet.FORTMATIC:
+      return fortmatic
+    case Wallet.NETWORK:
+      return network
+    case Wallet.GNOSIS_SAFE:
+      return gnosisSafe
+  }
+}
+
+const getConnectorListItemForWallet = (wallet: Wallet) => {
+  return {
+    connector: getConnectorForWallet(wallet),
+    hooks: getHooksForWallet(wallet),
+  }
+}
+
+export const getHooksForWallet = (wallet: Wallet) => {
+  switch (wallet) {
+    case Wallet.INJECTED:
+      return injectedHooks
+    case Wallet.COINBASE_WALLET:
+      return coinbaseWalletHooks
+    case Wallet.WALLET_CONNECT:
+      return walletConnectHooks
+    case Wallet.FORTMATIC:
+      return fortmaticHooks
+    case Wallet.NETWORK:
+      return networkHooks
+    case Wallet.GNOSIS_SAFE:
+      return gnosisSafeHooks
+  }
+}
+
+export const [network, networkHooks] = initializeConnector<Network>(
```

Comment:
A lot of boilerplate could be simplified by grouping these into a single object and referencing that object everywhere. eg

```
function createConnector<T>(initializer: PassThroughToInitializeConnector) {
  const [connector, hooks] = initializeConnector<T>(initializer)
  return { connector, hooks }
}
```

Then you can `export const network = createConnector<Network>(...` and use `network.connector` and `network.hooks` throughout (eg for `WalletModal/index.tsx`, as well as returning the full object from `getConnectorFromWallet`, and using this new object directly for your ordered connectors list).